evap/staff/tests/test_views.py

```diff
@@ -136,6 +136,92 @@ def test_deletes_users(self):
+        mommy.make(UserProfile, username="staff", groups=[Group.objects.get(name="Staff")])
+
+    def test_import_valid_file(self):
+        page = self.app.get(self.url, user='staff')
+
+        original_user_count = UserProfile.objects.count()
+
+        form = page.forms["user-import-form"]
+        form["excel_file"] = (self.filename_valid,)
+        page = form.submit(name="operation", value="test")
+
+        self.assertContains(page, 'Import previously uploaded file')
+        self.assertEqual(UserProfile.objects.count(), original_user_count)
+
+        form = page.forms["user-import-form"]
+        form.submit(name="operation", value="import")
+        self.assertEqual(UserProfile.objects.count(), original_user_count + 2)
+
+        page = self.app.get(self.url, user='staff')
+        self.assertNotContains(page, 'Import previously uploaded file')
+
+    def test_import_invalid_file(self):
+        page = self.app.get(self.url, user='staff')
+
+        original_user_count = UserProfile.objects.count()
+
+        form = page.forms["user-import-form"]
+        form["excel_file"] = (self.filename_invalid,)
+
+        reply = form.submit(name="operation", value="test")
+
+        self.assertContains(reply, 'Sheet &quot;Sheet1&quot;, row 2: Email address is missing.')
+        self.assertContains(reply, 'Errors occurred while parsing the input data. No data was imported.')
+        self.assertNotContains(reply, 'Import previously uploaded file')
+
+        self.assertEqual(UserProfile.objects.count(), original_user_count)
+
+    def test_import_wrong_file(self):
+        page = self.app.get(self.url, user='staff')
+
+        form = page.forms["user-import-form"]
+        form["excel_file"] = (self.filename_random,)
+
+        reply = form.submit(name="operation", value="test")
+
+        self.assertContains(reply, "Couldn&#39;t read the file. Error: Unsupported format, or corrupt file: Expected BOF record;")
+        self.assertNotContains(reply, 'Import previously uploaded file')
+
+    def test_suspicious_operation(self):
+        page = self.app.get(self.url, user='staff')
+
+        form = page.forms["user-import-form"]
+        form["excel_file"] = (self.filename_valid,)
+
+        # Should throw SuspiciousOperation Exception.
+        reply = form.submit(name="operation", value="hackit", expect_errors=True)
+
+        self.assertEqual(reply.status_code, 400)
+
+    def test_invalid_upload_operation(self):
+        page = self.app.get(self.url, user='staff')
+
+        form = page.forms["user-import-form"]
+        page = form.submit(name="operation", value="test")
+
+        self.assertContains(page, 'Please select an Excel file')
+        self.assertNotContains(page, 'Import previously uploaded file')
+
+    def test_invalid_import_operation(self):
+        page = self.app.get(self.url, user='staff')
+
+        form = page.forms["user-import-form"]
+        reply = form.submit(name="operation", value="import", expect_errors=True)
```

Comment:
No excel file was uploaded and tested previously. This POST request could only happen with malicious intention.